NGitLab/Impl/IGitLabCollectionResponse.cs

```diff
@@ -0,0 +1,8 @@
+﻿using System.Collections.Generic;
+
+namespace NGitLab
+{
+    public interface IGitLabCollectionResponse<T> : IEnumerable<T>, IAsyncEnumerable<T>
```

Comment:
You want to keep is sync too?

Reply:
I wonder if this interface is useful. There are two locations where we need to instantiate the collections: NGitLab (actual HTTP requests) and NGitLab.Mocks (using a collection). We could remove the interface and support both scenarios using two ctors.

Note that exposing a type instead of `IAsyncEnumerable` is useful for the future. Indeed, we got requests to query all pages in parallel or to know the number of items (accessible in the headers of the first requests). Using a specific type would allow adding more properties / methods in the future.